### DISCUSSION TITLE
自定义字段类型与formily类型的映射可否实现？

### DISCUSSION THREAD

**Discussion by rayy-li** [alibaba/formily#1932] [Q&A 2.x] [answered]
目前 Schema 字段类型属性 type 只能为 array 时创建为 ArrayField，object 是为 ObjectField，能不能够自定义类型映射成 ArrayField 和 ObjectField 字段类型？

**Comment by janryWang** [comment, marked as answer]
可以，看Schema文档

**Reply by rayy-li** [reply]
目前看 @formily/vue 中的 SchemaField 和 RecursionField 源码，是写死了判断 object 、 array 和 void 的类型判断~没有实现自定义类型映射的

**Reply by janryWang** [reply]
是Schema的静态方法注册映射的……

**Reply by rayy-li** [reply]
还是看不到是哪个方法
- registerVoidComponents 用于注册字段为 void 类型；
- registerTypeDefaultComponents 用于schema类型映射默认组件。
有没有类似 registerArrayComponents registerObjectComponents 的静态方法

**Reply by janryWang** [reply]
哦，看來我理解錯了，我一直以為你需要registerTypeDefaultComponents……然后你这个需求啊，具体是啥场景需要这个能力呢，目前react vue都不支持

**Reply by rayy-li** [reply]
场景是这样：公司项目中产品和后端使用了”实体关系模型“来描述业务和获取数据，那前端也想相应配合有 实体 类型https://doc.cuba-platform.cn/frontend/cuba-frontend-docs/v20.1/typescript-sdk/index.html
，这样前端开发就能基于这个实体类描述（元数据）来开发页面，落实到json-schema就是 name: 'xxx', type: 'entity' 属性知道需要怎样获取数据了

**Reply by janryWang** [reply]
那你可以写个patch，自己做转换，其实就是把特殊类型转换成普通类型
